In [1]:
import pandas as pd
import seaborn as sns
import requests

### Getting prices from Spain government API:

In [2]:
url_productos="https://sedeaplicaciones.minetur.gob.es/ServiciosRESTCarburantes/PreciosCarburantes/Listados/ProductosPetroliferos/"

In [3]:
response = requests.get(url=url_productos)

In [4]:
products_response = response.json()

In [5]:
products = pd.DataFrame(products_response)

In [6]:
products

IDProducto                NombreProducto NombreProductoAbreviatura
0           1                Gasolina 95 E5                     G95E5
1          23               Gasolina 95 E10                    G95E10
2          20        Gasolina 95 E5 Premium                    G95E5+
3           3                Gasolina 98 E5                     G98E5
4          21               Gasolina 98 E10                    G98E10
5           4            Gasóleo A habitual                       GOA
6           5               Gasóleo Premium                      GOA+
7           6                     Gasóleo B                       GOB
8           7                     Gasóleo C                       GOC
9          16                     Bioetanol                       BIE
10          8                     Biodiésel                       BIO
11         17   Gases licuados del petróleo                       GLP
12         18        Gas natural comprimido                       GNC
13         19           Gas natural licuado                       GNL
14         22                     Hidrógeno                        H2
15          9   Fuelóleo bajo índice azufre                       FOB
16         10             Fuelóleo especial                       FOE
17         11     Gasóleo para uso marítimo                       MGO
18         12          Gasolina de aviación                      GNAV
19         13  Queroseno de aviación JET_A1                     JETA1
20         14  Queroseno de aviación JET_A2                     JETA2

In [7]:
products[products.NombreProducto == "Gasolina 98 E10"]

IDProducto   NombreProducto NombreProductoAbreviatura
4         21  Gasolina 98 E10                    G98E10

#### This is the gas that we want to know the price within Spain Territory. Let's save it's product ID:

In [8]:
E10 = str(int(products[products.NombreProducto == "Gasolina 98 E10"].IDProducto))

#### Let's make a new API request with the prices:

In [23]:
url_prices = "https://sedeaplicaciones.minetur.gob.es/ServiciosRESTCarburantes/PreciosCarburantes/EstacionesTerrestres/FiltroProducto/"

In [24]:
E10_response = requests.get(url=url_prices+E10)

In [25]:
E10_df = pd.DataFrame(E10_response.json()["ListaEESSPrecio"])

In [26]:
E10_df

C.P.                             Dirección           Horario    Latitud  \
0  12160          CTRA. RAMAL A CULLA km 3,900  L-D: 08:00-21:00  40,379667   
1  18360        CARRETERA DE LA ESTACION KM. 1  L-D: 06:30-22:30  37,198778   
2  21610                       CR N-435, 219,9  L-D: 07:00-23:00  37,315444   
3  24400      CALLE BURDEOS  EN  PONFERRADA, 3          L-D: 24H  42,554389   
4  45830  CARRETERA QUINTANAR-ALCAZAR KM. 7600          L-D: 24H  39,525528   

             Localidad Longitud (WGS84) Margen            Municipio  \
0              BENASAL        -0,139500      D              Benasal   
1         HUETOR-TAJAR        -4,056806      N         Huétor Tájar   
2  SAN JUAN DEL PUERTO        -6,827944      I  San Juan del Puerto   
3           PONFERRADA        -6,613417      D           Ponferrada   
4       MIGUEL ESTEBAN        -3,080750      N       Miguel Esteban   

  PrecioProducto             Provincia Remisión               Rótulo  \
0          1,475  CASTELLÓN / CASTELLÓ       dm  CARCELLER ROYO S.L.   
1          1,519               GRANADA       dm   COMBUSTIBLES GOMEZ   
2          1,369                HUELVA       dm               COLOSO   
3          1,400                  LEÓN       dm            ZEPXA ECO   
4          1,498                TOLEDO       dm              NINGUNO   

  Tipo Venta IDEESS IDMunicipio IDProvincia IDCCAA  
0          P   9338        1829          12     10  
1          P   9640        2756          18     01  
2          P   5742        3278          21     01  
3          P  14909        3699          24     08  
4          P   5435        6861          45     07

#### It's seems like in Spain there's no many E10 gas stations.

#### Let's get and average in E10 gas in Spain:

In [27]:
E10_df.PrecioProducto = E10_df.PrecioProducto.str.replace(",", ".")

In [30]:
E10_df.PrecioProducto = E10_df.PrecioProducto.astype(float)

In [32]:
E10_df.PrecioProducto.mean()

1.4522

#### Now we have it! The mean price of E10 gas in Spain is 1.4522 euros/lt.

#### If we have more gas stations in Spain selling E10 i could make a better analysis between each community of Spain and their prices.